clients/java/src/main/java/io/carbynestack/thymus/client/ThymusVCClient.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2024 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository https://github.com/carbynestack/thymus.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.carbynestack.thymus.client;
+
+import io.vavr.Tuple;
+import io.vavr.Tuple2;
+import io.vavr.collection.Stream;
+import io.vavr.concurrent.Future;
+import io.vavr.control.Either;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+import java.util.*;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+/**
+ * A Thymus client to interact with all the Thymus services of a Carbyne Stack
+ * Virtual Cloud.
+ */
+@Slf4j
+public class ThymusVCClient {
+
+    private final List<ThymusVCPClient> vcpClients;
+    private final Executor executor;
+
+    ThymusVCClient(List<ThymusVCPClient> vcpClients) {
+        if (vcpClients.isEmpty()) {
+            throw new IllegalArgumentException("At least one VCP client must be provided");
+        }
+        this.vcpClients = vcpClients;
+        this.executor = Executors.newFixedThreadPool(vcpClients.size());
+    }
+
+    /**
+     * The endpoints this client talks to.
+     *
+     * @return The list of endpoints
+     */
+    public List<ThymusEndpoint> getThymusEndpoints() {
+        return vcpClients.stream().map(ThymusVCPClient::getEndpoint).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the policies available within the Carbyne Stack Virtual Cloud.
+     *
+     * @return The intersection of the policies registered across all the VCPs in the VC.
+     */
+    public Future<Either<ThymusError, Set<NamespacedName>>> getPolicies() {
+        // Execute all the requests in parallel
+        Stream<Future<Tuple2<ThymusEndpoint, Either<ThymusError, List<NamespacedName>>>>> invocations =
+                Stream.ofAll(vcpClients).map(c -> Future.of(executor, () -> {
+                    log.info("Fetching policies from endpoint {}", c.getEndpoint());
+                    return Tuple.of(c.getEndpoint(), c.getPolicies());
+                }));
+
+        // Collect the results and translate into a single aggregate result
+        return Future.sequence(invocations).map(results -> {
+            Set<ThymusError> errors = new HashSet<>();
+            Set<NamespacedName> commonPolicies = new HashSet<>();
+            results.zipWithIndex().forEach(r -> {
+                val index = r._2;
+                val result = r._1;
+                if (result._2.isLeft()) {
+                    // Collect all errors indexed by the endpoint
+                    errors.add(result._2.getLeft());
+                } else {
+                    // Compute the intersection of all policies
+                    if (index == 0) {
+                        commonPolicies.addAll(result._2.get());
+                    } else {
+                        commonPolicies.retainAll(result._2.get());
```

Comment:
It is ignored if at least one provider has policies defined that are not present on other(s).
Should that be logged somehow? Is this considered okay? 
It could be a common scenario if policies are updated (added or deleted) and changes are not yet reflected on all VCPs. However, an option to have this information printed allows debugging if policies are not enrolled properly